### PR TITLE
Add deprecation messages

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Comment by howardjohn** [comment]
@ostromart @richardwxn looking for some advice here.

If I keep data/profiles/default.yaml using the deprecated options, then it will output a ton of errors about deprecated usages, so I moved them. However, `values.meshConfig` takes precedence over the deprecated settings. Therefor the profile is actually overriding the user settings.

Any istioctl fix I can make here, or should I make it so that the defaults are specified in the charts instead of as part of the profile?

This is why the security test fails, by the way. The IOP:
```yaml

apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  hub: istio-testing
  tag: istio-testing
  
  values:
    global:
      certificates:
        - dnsNames: [istio-pilot.istio-system.svc, istio-pilot.istio-system]
        - secretName: dns.istio-galley-service-account
          dnsNames: [istio-galley.istio-system.svc, istio-galley.istio-system]
        - secretName: dns.istio-sidecar-injector-service-account
          dnsNames: [istio-sidecar-injector.istio-system.svc, istio-sidecar-injector.istio-system]
```

But `values.meshConfig.certificates=[]` from default profile overrides this

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go [hunk 1]
```diff
@@ -42,9 +43,53 @@ func ValidateConfig(failOnMissingValidation bool, iopvalues map[string]interface
...
 }
 
+func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
+	global := values.GetGlobal()
+	pc := global.GetProxy()
+	messages := []string{}
+	addDeprecation := func(old, new string) {
+		messages = append(messages, fmt.Sprintf("! %s is deprecated; use %s", old, new))
+	}
+	if global.GetCertificates() != nil {
```
Comment (ostromart) [inline operator/pkg/apis/istio/v1alpha1/validation/validation.go:57]: You can use tpath.GetFromStructPath and a table to express this a bit more compactly.

Comment (howardjohn) [reply operator/pkg/apis/istio/v1alpha1/validation/validation.go]: I agree in general but it doesn't handle ints/durations/strings that are not nil. I guess I can probably workaround that though, thanks for the suggestion. Can you take a look at the workaround (default values) to see if there is a better one?

**Comment by howardjohn** [comment]
I fixed the override values by putting the defaults in the chart directly

File: operator/data/profiles/default.yaml [hunk 2]
```diff
@@ -251,6 +244,11 @@ spec:
   # Global values passed through to helm global.yaml.
   # Please keep this in sync with manifests/global.yaml
   values:
+    # You may override parts of meshconfig by uncommenting the following lines.
+    meshConfig:
```
Comment (richardwxn) [inline operator/data/profiles/default.yaml:248]: why we need to move to values?

Comment (howardjohn) [reply operator/data/profiles/default.yaml:248]: Well the current top level field is broken. If we fix that I suppose we can undo this, and then the behavior will be identical

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go [hunk 3]
```diff
@@ -42,9 +45,50 @@ func ValidateConfig(failOnMissingValidation bool, iopvalues map[string]interface
...
+	warningMessage += deprecatedSettingsMessage(values)
 	return validationErrors, warningMessage
 }
 
+// Converts from helm paths to struct paths
+// global.proxy.accessLogFormat -> Global.Proxy.AccessLogFormat
+func firstCharsToUpper(s string) string {
+	res := []string{}
+	for _, ss := range strings.Split(s, ".") {
+		res = append(res, strings.ToUpper(ss[0:1])+ss[1:])
```
Comment (morvencao) [inline operator/pkg/apis/istio/v1alpha1/validation/validation.go:57]: nit: it's better to to use `strings.Title(ss)`

**Review by ostromart — APPROVED** [review]

**Pull request merged**